aten/src/ATen/native/TensorIterator.h

```diff
@@ -102,11 +103,13 @@ struct CAFFE2_API OperandInfo {
   /// specifies which type to allocate. Note that there is very limited support
   /// for type conversions currently: they are only allowed for zero-dim tensors.
   Device device = kCPU;
-  ScalarType dtype = ScalarType::Undefined;
+  ScalarType target_dtype = ScalarType::Undefined;
+  // Caching dtype of the tensor, because scalar_type is an expensive operation
+  ScalarType tensor_dtype = ScalarType::Undefined;
```

Comment:
Can you please inline comments how `target_dtype` is different from `tensor_dtype`